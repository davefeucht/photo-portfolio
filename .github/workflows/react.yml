# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions

name: React CI/CD

on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]

jobs:
  build:

    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: [18.x]

    steps:
    - name: Checkout code
      uses: actions/checkout@v2

    - name: Install Node.js ${{ matrix.node-version }}
      uses: actions/setup-node@v1
      with:
        node-version: ${{ matrix.node-version }}

    - name: Enable corepack
      run: corepack enable

    - name: Install dependencies
      run: yarn install

    - name: Build project
      run: npm run build --if-present

    - name: Run tests
      run: npm test
