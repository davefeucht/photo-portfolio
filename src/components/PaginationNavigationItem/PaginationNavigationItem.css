.pagination-navigation__item {
    margin: 3px;
    color: #FFFFFF;
}

.pagination-navigation__item.selected {
    font-weight: bold;
    font-size: larger
}

.pagination-navigation__item.deselected {
    cursor: pointer;
}
