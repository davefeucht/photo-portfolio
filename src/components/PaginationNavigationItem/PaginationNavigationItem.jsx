/****************
* PaginationNavigationItem component displays one item of the pagination navigation
****************/

import React from "react";
import PropTypes from 'prop-types';
import './PaginationNavigationItem.css';

const PaginationNavigationItem = ({ content, selectedState, navigationFunction }) => {
    return (
        <div className={`pagination-navigation__item ${selectedState}`} onClick={navigationFunction ? () => navigationFunction(parseInt(content)) : undefined}>{content}</div>
    )
};

PaginationNavigationItem.displayName = 'PaginationNavigationItem';

PaginationNavigationItem.propTypes = {
    content: PropTypes.string.isRequired,
    selectedState: PropTypes.string.isRequired,
    navigationFunction: PropTypes.func
};

export default PaginationNavigationItem;
