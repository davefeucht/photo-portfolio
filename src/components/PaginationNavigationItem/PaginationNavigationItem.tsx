/** **************
* PaginationNavigationItem component displays one item of the pagination navigation
*************** */

import './PaginationNavigationItem.css';

import * as React from 'react';

interface PaginationNavigationItemProps {
    content: string,
    selectedState: string,
    navigationFunction(content: number): void,
    isLast: boolean
}

const PaginationNavigationItem: React.FC<PaginationNavigationItemProps> = ({
    content, selectedState, navigationFunction, isLast
}) => {
    return (
        <div
            className={`pagination-navigation__item ${selectedState}`}
            onClick={navigationFunction ? () => navigationFunction(parseInt(content)) : undefined}
        >
            {content}
            {!isLast ? ' â€¢ ' : ''}
        </div>
    );
};

PaginationNavigationItem.displayName = 'PaginationNavigationItem';

export default PaginationNavigationItem;
