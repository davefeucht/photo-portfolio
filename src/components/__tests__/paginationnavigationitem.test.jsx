import { render } from '@testing-library/react';
import React from 'react';
import { act } from 'react-dom/test-utils';
import { MemoryRouter } from 'react-router-dom';

import PaginationNavigationItem from '../PaginationNavigationItem/PaginationNavigationItem';

const content = '1';
const selectedState = 'selected';
const navigationFunction = () => {};
const isLast = false;

test('PaginationNavigationItem displays', async () => {
    let container;
    await act(async () => {
        container = render(
            <MemoryRouter>
                <PaginationNavigationItem content={content} selectedState={selectedState} navigationFunction={navigationFunction} isLast={isLast} />
            </MemoryRouter>
        ).container;
    });
    expect(container.firstChild).toMatchSnapshot();
});
