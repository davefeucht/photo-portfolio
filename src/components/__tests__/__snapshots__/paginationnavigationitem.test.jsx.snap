// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationNavigationItem displays 1`] = `
<div
  aria-label="pagination"
  class="pagination-navigation__item selected"
>
  1
   | 
</div>
`;
